[Chan Rui Jia] iP

matochichap opened the pull request:
### ChatGPA 5.0

> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes))

ChatGPA 5.0 frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/matochichap/ip/releases/tag/A-Jar)
2. double-click it
3. add your tasks
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (Coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) {
    new Duke("data/duke.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

matochichap committed:
Complete Level-0

matochichap committed:
Complete Level-1

matochichap committed:
Complete Level-2

matochichap committed:
Complete Level-3

matochichap committed:
Complete Level-4

matochichap committed:
Complete A-TextUiTesting

matochichap committed:
Complete Level-5

matochichap committed:
Complete Level-6

matochichap committed:
Complete A-Enums

matochichap committed:
Complete Level-7

matochichap committed:
Revert "Complete A-Enums"

This reverts commit 5a1477f6ffa519d5d35b4fe496c195b46c098817.

matochichap committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Action.java
#	src/main/java/duke/ChatManager.java

matochichap committed:
Complete Level-8

matochichap committed:
Merge branch 'branch-level-8'

matochichap committed:
Complete A-MoreOOP and A-Packages

matochichap committed:
Move Duke.java into duke package

matochichap committed:
Merge branch 'add-gradle-support'

matochichap committed:
Edit build gradle

matochichap committed:
Complete A-JUnit

matochichap committed:
Edit build.gradle

matochichap committed:
Complete A-JavaDoc

matochichap committed:
Complete A-CodingStandard

matochichap committed:
Complete Level-9

matochichap committed:
Merge branch 'branch-A-JavaDoc'

matochichap committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/duke/message/AddTaskMessage.java
#	src/main/java/duke/message/DeleteTaskMessage.java
#	src/main/java/duke/message/ErrorMessage.java
#	src/main/java/duke/message/MarkTaskMessage.java
#	src/main/java/duke/message/Message.java
#	src/main/java/duke/message/TaskListMessage.java
#	src/main/java/duke/message/UnmarkTaskMessage.java
#	src/main/java/duke/parser/DateTimeParser.java
#	src/main/java/duke/parser/FileParser.java
#	src/main/java/duke/parser/UserInputParser.java
#	src/main/java/duke/task/DeadlinesTask.java
#	src/main/java/duke/task/EventsTask.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/TodoTask.java

matochichap committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/TaskList.java

matochichap committed:
Complete A-CheckStyle

matochichap committed:
Merge branch 'branch-A-CheckStyle'

matochichap committed:
Complete Level-10

matochichap committed:
Merge branch 'branch-level-10'

matochichap committed:
Use FXML

bhnuka commented on src/main/java/duke/gui/DialogBox.java:
```diff
@@ -0,0 +1,65 @@
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    public void setDialog(String text) {
```
Would be good to add Javadoc comments! But I understand that this this applies mostly for recently committed code + you only have to do minimally 50% of Javadoc comments so understandable :)

bhnuka commented on src/main/java/duke/gui/MainWindow.java:
```diff
@@ -0,0 +1,77 @@
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
```
Would be good to break this up into two lines of comment (113 characters). While within the hard limit of 120 characters, having the comment as two lines would help readability immensely!

bhnuka commented on src/main/java/duke/parser/FileParser.java:
```diff
@@ -0,0 +1,58 @@
+            String to;
+            boolean isDone = !mark.equals("0");
+            switch (type) {
+            case "T":
```
Following the Java coding standards! Excellent 😄

bhnuka commented on src/main/java/duke/parser/UserInputParser.java:
```diff
@@ -0,0 +1,144 @@
+            name = userInput.split(" ", 2)[1];
+            return taskList.add(new TodoTask(name, false));
+        case DEADLINE:
+            // TODO: fix bug and test
```
Great that you add this here as a reminder 👍

bhnuka reviewed (commented):
Overall, you adhere very strongly to the coding standard. LGTM. A few nits (that aren't really nits, more of things I'm sure you would've gotten around to eventually). Keep up the great work!

sunzihan23 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,58 @@
+import duke.templates.MessageTemplates;
+
+/**
+ * Represents the Storage.
```
Can consider writing a slightly more detailed JavaDoc for all the codes?

sunzihan23 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,92 @@
+
+    /**
+     * Main method.
+     * @param args Arguments.
```
Would be good to have an empty line between description and parameter sections, according to the JavaDoc guide!

matochichap committed:
Complete A-Assertions

Fix bugs for invalid user input causing index error in deadline and
event case.

Use assert as a double check in case of changes to the code.

Minor changes for other classes.

matochichap committed:
Merge pull request #1 from matochichap/branch-A-Assertions

Complete A-Assertions

matochichap committed:
Complete A-CodeQuality

Some parts of code do not adhere to coding standards.

Fixed parts of code to adhere to coding standards and improve
readability.

matochichap committed:
Merge pull request #2 from matochichap/branch-A-CodeQuality

Complete A-CodeQuality

matochichap committed:
Complete A-Streams

Change a for loop to use streams to fulfill requirement for A-Streams.

matochichap committed:
Merge pull request #3 from matochichap/branch-A-Streams

Complete A-Streams

matochichap committed:
Complete C-Help

matochichap committed:
Merge branch 'branch-C-Help'

matochichap committed:
Complete A-CI

matochichap committed:
Complete A-CI

matochichap committed:
Revert "Complete A-CI"

This reverts commit 1f178c3f3e7d5b491faa47bb4cfd5dee357aa016.

matochichap committed:
Merge branch 'master' of https://github.com/matochichap/ip

matochichap committed:
Edit gradle.yml

matochichap committed:
Revert "Edit gradle.yml"

This reverts commit 1e5e45230ff61555c7fd8372bda9f23a52bf2806.

matochichap committed:
Update gradle.yml

matochichap committed:
Merge branch 'master' of https://github.com/matochichap/ip

matochichap committed:
Update gradle.yml

matochichap committed:
Fix documentation

Fix documentation to adhere to coding standards.

Give credit for reusing GUI tutorial code.

matochichap committed:
Complete A-UserGuide

matochichap committed:
Merge branch 'branch-A-UserGuide'

matochichap committed:
Update README.md

Update the correct README file this time.

matochichap committed:
Update README.md

matochichap committed:
Update README.md

matochichap committed:
Update README.md

matochichap committed:
update .yml

matochichap committed:
update .yml
